shell/platform/windows/flutter_windows_view_unittests.cc

```diff
@@ -94,15 +103,57 @@ std::unique_ptr<FlutterWindowsEngine> GetTestEngine() {
   return engine;
 }
 
+class MockFlutterWindowsEngine : public FlutterWindowsEngine {
+ public:
+  MockFlutterWindowsEngine() : FlutterWindowsEngine(GetTestProject()) {}
+
+  MOCK_METHOD0(Stop, bool());
+
+ private:
+  FML_DISALLOW_COPY_AND_ASSIGN(MockFlutterWindowsEngine);
+};
+
+class MockAngleSurfaceManager : public AngleSurfaceManager {
+ public:
+  MockAngleSurfaceManager() {}
+
+  MOCK_METHOD0(DestroySurface, void());
+
+ private:
+  FML_DISALLOW_COPY_AND_ASSIGN(MockAngleSurfaceManager);
+};
+
 }  // namespace
 
+// The view's surface must be destroyed after the engine is shutdown.
+// See: https://github.com/flutter/flutter/issues/124463
+TEST(FlutterWindowsViewTest, Shutdown) {
+  std::unique_ptr<MockFlutterWindowsEngine> engine =
+      std::make_unique<MockFlutterWindowsEngine>();
+  auto window_binding_handler =
+      std::make_unique<NiceMock<MockWindowBindingHandler>>();
+  std::unique_ptr<MockAngleSurfaceManager> surface_manager =
+      std::make_unique<MockAngleSurfaceManager>();
+
+  EngineModifier modifier(engine.get());
+  FlutterWindowsView view(std::move(window_binding_handler));
+
+  // The engine must be stopped before the surface can be destroyed.
+  InSequence s;
+  EXPECT_CALL(*engine.get(), Stop);
```

Comment:
Minor nit: Should this specify a `Times`?

Reply:
Good idea, updated!